src/geode/mesh/core/aabb_helpers.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2019 - 2020 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+/*
+ * Modified from RINGMesh https://github.com/ringmesh/RINGMesh
+ * Copyright (c) 2012-2018, Association Scientifique pour la Geologie et ses
+ * Applications (ASGA)
+ */
+
+#include <geode/mesh/core/aabb_helpers.h>
+
+#include <memory>
+
+#include <geode/mesh/core/edged_curve.h>
+#include <geode/mesh/core/triangulated_surface.h>
+
+#include <geode/geometry/aabb.h>
+#include <geode/geometry/basic_objects.h>
+#include <geode/geometry/distance.h>
+#include <geode/geometry/point.h>
+#include <geode/geometry/vector.h>
+
+namespace geode
+{
+    template < index_t dimension >
+    AABBTree< dimension > create_aabb_tree(
+        const EdgedCurve< dimension >& mesh )
+    {
+        absl::FixedArray< BoundingBox< dimension > > box_vector(
+            mesh.nb_edges() );
+        for( const auto e : Range{ mesh.nb_edges() } )
+        {
+            box_vector[e].add_point(
+                mesh.point( mesh.edge_vertex( { e, 0 } ) ) );
+            box_vector[e].add_point(
+                mesh.point( mesh.edge_vertex( { e, 1 } ) ) );
+        }
+        return AABBTree< dimension >( box_vector );
```

Comment:
Use {} when creating an object as much as possible. Same comment at several location in the file.

Reply:
should I write         
absl::FixedArray< BoundingBox< dimension > > box_vector{
            mesh.nb_edges()
        };
as well... I do not like the way clang is formatting this